### ISSUE TITLE
During kubernetes version upgrade node_locations is problematic

### ISSUE THREAD

**Issue by jgacek-ttpsc** [terraform-google-modules/terraform-google-kubernetes-engine#1202]
### TL;DR

During the terraform code deployment with kubernetes_version changed to new version it breaks. It's the same release channel.

### Expected behavior

It should upgrade the cluster to selected version.

### Observed behavior

 Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for module.cluster.module.gke.google_container_cluster.primary to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/google" produced an invalid new value for .node_locations: actual set element cty.StringVal("us-central1-a") does not correlate with any element in plan.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.

### Terraform Configuration

```hcl
module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google"
  version                    = ">=17.2.0"
  project_id                 = var.project_id
  name                       = "${var.deployment_name}-deployment"
  region                     = var.region
  zones                      = data.google_compute_zones.available.names
  network                    = var.network_name
  subnetwork                 = "${var.deployment_name}-deployment-main"
  ip_range_pods              = "pods"
  ip_range_services          = "services"
  http_load_balancing        = false
  horizontal_pod_autoscaling = true
  network_policy             = false
  create_service_account    = false
  service_account           = var.service_accounts_email
  # release_channel           = "STABLE" <- before upgrade
  kubernetes_version        = "1.21.5-gke.1805" <- tried this version on upgrade
  cluster_autoscaling       = {
    enabled = true
    min_cpu_cores = 15
    max_cpu_cores = 300
    min_memory_gb = 20
    max_memory_gb = 600
    gpu_resources = []
  }

  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
    
  }

  node_pools_labels = {
    all = {
      "${var.deployment_name}" = true
    }
  }
}
```

### Terraform Version

```sh
1.1.7
```

### Additional information

Version 20.0.0 of module. Probably issue with shuffle_random on node_locations.
I couldn't set node_locations because it's predefined in locals.
